filters/include/pcl/filters/spherical_voxel_grid.h

```diff
@@ -0,0 +1,357 @@
+        * \param[in] downsample the new value (true/false)
+        */
+      inline void
+      setDownsampleAllData (bool downsample) { downsample_all_data_ = downsample; }
+
+      /** \brief Get the state of the internal downsampling parameter (true if
+        * all fields need to be downsampled, false if just XYZ).
+        */
+      inline bool
+      getDownsampleAllData () { return (downsample_all_data_); }
+
+      /** \brief Set the minimum number of points required for a voxel to be used.
+        * \param[in] min_points_per_voxel the minimum number of points for required for a voxel to be used
+        */
+      inline void
+      setMinimumPointsNumberPerVoxel (unsigned int min_points_per_voxel) { min_points_per_voxel_ = min_points_per_voxel; }
+
+      /** \brief Return the minimum number of points required for a voxel to be used.
+        */
+      inline unsigned int
+      getMinimumPointsNumberPerVoxel () { return (min_points_per_voxel_); }
+
+      /** \brief Return the maximum radius of a point from the ceter,
+        * i.e. the radius of the spherical voxel grid (after filtering)
+        */
+      inline float
+      getMaximumRadius () { return (max_radius_); }
+
+      /** \brief Set the cartesian origin used to build the spherical grid.
+        * \param[in] origin the cartesian origin
+        */
+      inline void
+      setOrigin (const Eigen::Vector4f& origin) { filter_origin_ = origin; }
+
+      /** \brief Get the cartesian origin used to build the spherical grid.
+        */
+      inline Eigen::Vector3f
+      getOrigin () { return (filter_origin_.head<3>()); }
+
+      /** \brief Provide the name of the field to be used for filtering data. In conjunction with  \a setFilterLimits,
+        * points having values outside this interval will be discarded.
+        * \param[in] field_name the name of the field that contains values used for filtering
+        */
+      inline void
+      setFilterFieldName (const std::string &field_name) { filter_field_name_ = field_name; }
+
+      /** \brief Get the name of the field used for filtering. */
+      inline std::string const
+      getFilterFieldName () { return (filter_field_name_); }
+
+      /** \brief Set the field filter limits. All points having field values outside this interval will be discarded.
+        * \param[in] limit_min the minimum allowed field value
+        * \param[in] limit_max the maximum allowed field value
+        */
+      inline void
+      setFilterLimits (const double &limit_min, const double &limit_max)
+      {
+        filter_limit_min_ = limit_min;
+        filter_limit_max_ = limit_max;
+      }
+
+      /** \brief Get the field filter limits (min/max) set by the user. The default values are -FLT_MAX, FLT_MAX.
+        * \param[out] limit_min the minimum allowed field value
+        * \param[out] limit_max the maximum allowed field value
+        */
+      inline void
+      getFilterLimits (double &limit_min, double &limit_max)
```

Comment:
Did you choose API for this function following some other similar existing in PCL? Then let's keep it this way for consistency, otherwise I'd propose to return `std::pair<double, double>`.

Reply:
Yes, this is the same filter field interface that `VoxelGrid` uses.

Reply:
OK, let's keep it this way then.